pkg/chezmoi/chezmoi.go

```diff
@@ -298,7 +298,7 @@ func etcHostsFQDNHostname(fileSystem vfs.FS) (string, error) {
 		text, _, _ = strings.Cut(text, "#")
 		fields := whitespaceRx.Split(text, -1)
 		if len(fields) > 1 && net.ParseIP(fields[0]).IsLoopback() &&
-			strings.Contains(fields[1], ".") {
+			strings.Contains(fields[1], ".") && fields[1] != "localhost.localdomain" {
```

Comment:
I would maybe make this `!strings.StartsWith(fields[1], "localhost")`. It would break if the name is `localhostfoo`, but it would prevent the case of `localhost.localdomain localhost` or `localhost localhost.localdomain`. I also think that `localdomain` may be a Linux thing; it could be `localhost.local` for macOS.

Reply:
Good point -- thanks! I've updated the PR to explicitly exclude `.localdomain` and `.local`.